BootloaderCommonPkg/Library/IppCryptoLib/rsa_verify.c

```diff
@@ -146,17 +146,173 @@ int VerifyRsaSignature (CONST PUB_KEY_HDR *PubKeyHdr, CONST SIGNATURE_HDR *Signa
 }
 
 
-/* Wrapper function for RSAVerify to make the inferface consistent.
+/* Wrapper function for RSA PSS verify to make the inferface consistent.
+ * Returns non-zero on failure, 0 on success.
+ */
+int VerifyRsaPssSignature (CONST PUB_KEY_HDR *PubKeyHdr, CONST SIGNATURE_HDR *SignatureHdr,  CONST UINT8  *Src, CONST UINT32  Size)

```

Comment:
Have we done some negative testing for this new method?
What happens when we manually corrupt a PSS signature in the stitched SBL image?
Does the implementation gracefully returns failure and block the boot?

Reply:
Yes, checked with corrupting the CfgData and PSS verification fails with an graceful exit.
RSA verification used for ext components -
CFG Data, KeyhashManifest, external component loaded using container and Boot images.
Boot halts only when boot image verification fails. For rest of the components, boot would continue.  Corresponding features would fails.

Reply:
Thanks Subash for running these tests.